src/BTree.h

```diff
@@ -1892,8 +1909,10 @@ class btree {
      * Clears this tree.
      */
     void clear() {
-        delete root;
-
+        numElements = 0;
+        if (root) {
```

Comment:
if (root != nullptr)

or just delete it, since deleting nullptr is defined to have no effect